lib/ecto/query/builder.ex

```diff
@@ -188,6 +188,12 @@ defmodule Ecto.Query.Builder do
     {expr, {[expr | params], acc}}
   end
 
+  # dynamic
+  def escape({:dynamic, _, [expr]}, _, {params, acc}, _vars, _env) do
+    acc = %{acc | expand_dynamic?: true}
```

Comment:
Should we check if `acc` has a `expand_dynamic?` flag and, if not, we raise saying the `type` does not support `dynamic`?